x-pack/platform/plugins/shared/alerting_v2/server/rule_executor/task_definition.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { CoreStart, Logger } from '@kbn/core/server';
+import type { TaskManagerSetupContract } from '@kbn/task-manager-plugin/server';
+import { schema } from '@kbn/config-schema';
+
+import type { AlertingV2Config } from '../config';
+import type { AlertingServerStartDependencies } from '../types';
+import { createRuleExecutorTaskRunner } from './task_runner';
+
+export const ALERTING_RULE_EXECUTOR_TASK_TYPE = 'alerting:esql' as const;
+
+export function initializeRuleExecutorTaskDefinition(
+  logger: Logger,
+  taskManager: TaskManagerSetupContract,
+  coreStartServices: Promise<[CoreStart, AlertingServerStartDependencies, unknown]>,
+  config: AlertingV2Config
+) {
+  taskManager.registerTaskDefinitions({
+    [ALERTING_RULE_EXECUTOR_TASK_TYPE]: {
+      title: 'Alerting v2 rule executor (ES|QL)',
+      timeout: '5m',
```

Comment:
I was thinking what would be a good value also for the director. If we process a lot of records, maybe we want this to be higher. Not sure tbh.